src/traits.rs

```diff
@@ -67,6 +67,40 @@ impl Enlargeable for u32 {
     type Larger = u64;
 }
 
+/// Linear interpolation without involving floating numbers.
+pub trait Lerp: Sized + Bounded + NumCast {
+    type SignedLarger: Primitive + AddAssign + Neg + 'static;
+
+    fn lerp(a: Self, b: Self, i: Self::SignedLarger, n: Self::SignedLarger) -> Self {
```

Comment:
This seems counter-intuitive with the fact that we want to use floating points in order to avoid overflows. Integers have been initially chosen in order to be more performant, but it can lead to overflows if not handled properly as you mentioned in the review.

Reply:
Seems fine as well.